plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -329,19 +332,142 @@ int CppParser::worker(const clang::tooling::CompileCommand& command_)
+            {
+              // Delete CppEntity
+              auto delCppEntities = _ctx.db->query<model::CppEntity>(
+                odb::query<model::CppEntity>::astNodeId == astNode.id);
+              for (const model::CppEntity &entity : delCppEntities)
+              {
+                _ctx.db->erase<model::CppEntity>(entity.id);
+              }
+
+              // Delete CppInheritance
+              auto delCppInheritance = _ctx.db->query<model::CppInheritance>(
+                odb::query<model::CppInheritance>::derived == astNode.mangledNameHash);
+              for (const model::CppInheritance &inheritance : delCppInheritance)
+              {
+                _ctx.db->erase<model::CppInheritance>(inheritance.id);
+              }
+
+              // Delete CppFriendship
+              auto delCppFriendship = _ctx.db->query<model::CppFriendship>(
+                odb::query<model::CppFriendship>::target == astNode.mangledNameHash);
+              for (const model::CppFriendship &friendship : delCppFriendship)
+              {
+                _ctx.db->erase<model::CppFriendship>(friendship.id);
+              }
+
+              // Delete CppNode (connected to CppAstNode)
+              auto delNodes = _ctx.db->query<model::CppNode>(
+                odb::query<model::CppNode>::domainId == std::to_string(astNode.id) &&
+                odb::query<model::CppNode>::domain == model::CppNode::CPPASTNODE);
+              for (model::CppNode &node : delNodes)
+              {
+                _ctx.db->erase<model::CppNode>(node.id);
+                // TODO: handle that RelationCollector will try to reinsert the other end of the relation
+                // Raises "[WARNING] object already persistent"
+              }
+            }
+
+            // Delete BuildAction
+            auto delSources = _ctx.db->query<model::BuildSource>(
+              odb::query<model::BuildSource>::file == delFile->id);
+            for (const model::BuildSource &source : delSources)
+            {
+              _ctx.db->erase<model::BuildAction>(source.action->id);
+            }
+
+            // Delete CppNode (connected to File)
+            auto delNodes = _ctx.db->query<model::CppNode>(
+              odb::query<model::CppNode>::domainId == std::to_string(delFile->id) &&
+              odb::query<model::CppNode>::domain == model::CppNode::FILE);
+            for (model::CppNode &node : delNodes)
+            {
+              _ctx.db->erase<model::CppNode>(node.id);
+              // TODO: handle that RelationCollector will try to reinsert the other end of the relation
+              // Raises "[WARNING] object already persistent"
+            }
+          }
+
+          case IncrementalStatus::ADDED:
+            // Empty deliberately
+            break;
+        }
+      }
+    }); // end of transaction
+  }
+  catch(odb::database_exception&)
+  {
+    LOG(fatal) << "Transaction failed!";
```

Comment:
`failed in C++ parser` instead, if `metrics` also mentions who they are.

Reply:
Fixed in 64cb407581662e78675baf916a95dba2b765b3e4.